src/App.tsx

```diff
@@ -1,31 +1,128 @@
 import React from 'react';
-import './App.scss';
+
 import './styles/general.scss';
-import { TodoList } from './components/TodoList';
+import './App.scss';
+import * as API from './api';
+import { TodoList, SortBy } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
 
 interface State {
   selectedUserId: number;
+  user: User | null;
+  searchQuery: string;
+  todos: Todo[];
+  filterCompleted: SortBy;
 }
 
 class App extends React.Component<{}, State> {
   state: State = {
     selectedUserId: 0,
+    user: null,
+    searchQuery: '',
+    todos: [],
+    filterCompleted: SortBy.default,
+  };
+
+  async componentDidMount() {
+    this.loadTodos();
+  }
+
+  loadUser = async (userId: number) => {
+    const user = await API.getUser(userId);
+
+    this.setState({
+      user,
+    });
+  };
+
+  loadTodos = async (filter = '') => {
+    const todos = await API.getTodos(filter);
+
+    this.setState({
+      todos,
+    });
+  };
+
+  handleSelectUser = (userId: number) => {
+    this.setState(() => ({
+      selectedUserId: userId,
+    }), () => this.loadUser(this.state.selectedUserId));
+  };
+
+  handleClearUser = () => {
+    this.setState(() => ({
+      user: null,
+      selectedUserId: 0,
+    }));
+  };
+
+  handleChangeSearchInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      searchQuery: event.target.value,
+    });
+  };
+
+  handleSelectCompletionChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState(() => ({
+      filterCompleted: event.target.value as SortBy,
+    }), () => this.loadTodos(this.state.filterCompleted));
+  };
+
+  getVisibleTodos = (todos: Todo[], str: string): Todo[] => {
+    return todos.filter(todo => todo.title.toLowerCase().includes(str.toLowerCase()));
```

Comment:
str.toLowerCase() - better to write in variable, for each `filter` iteration you make operation toLowerCase for str
with the separate variable, you will make toLowerCase only once